pkg/core/transaction/signer.go

```diff
@@ -86,3 +86,26 @@ func SignersToStackItem(signers []Signer) stackitem.Item {
 	}
 	return stackitem.NewArray(res)
 }
+
+// Copy creates a deep copy of the Signer.
+func (c *Signer) Copy() *Signer {
+	if c == nil {
+		return nil
+	}
+	signerCopy := *c
+	if c.AllowedContracts != nil {
+		signerCopy.AllowedContracts = make([]util.Uint160, len(c.AllowedContracts))
+		copy(signerCopy.AllowedContracts, c.AllowedContracts)
```

Comment:
@roman-khimov, for `util.Uint160` the question is the same, do we need the *deep* copy of every signer's hash?